src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -164,4 +188,22 @@ private function getTimezone(array $context)
 
         return $dateTimeZone instanceof \DateTimeZone ? $dateTimeZone : new \DateTimeZone($dateTimeZone);
     }
+
+    private function isPreserveContextTimezone(array $context): bool
+    {
+        //Version 5.0 of Symfony will always preserve the context timezone.
+        if (class_exists('\Symfony\Component\HttpKernel\Kernel') && version_compare(Kernel::VERSION, '5.0.00', '>=')) {
```

Comment:
You should rely on the Kernel version. http-kernel and serializer are two different components that can be installed in different versions. That's not reliable.

But actually, you don't need to do this.
Default will stay "keeping the input timezone if provided" in 4.4. In 5.0, we'll update the code directly to make it always using the context timezone. No condition should be done for this, only the code base updated.

Reply:
I guess I was trying to ensure that this code did not have to be revisited to ensure correct operation at Symfony 5.0, and I do not see anyway to read the version of the Serializer to ensure this code automatically retires.